src/typing_extensions.py

```diff
@@ -135,7 +135,7 @@ def _collect_type_vars(types, typevar_types=None):
         ):
             tvars.append(t)
         if _should_collect_from_parameters(t):
-            tvars.extend([t for t in t.__parameters__ if t not in tvars])
+            tvars.extend(t for t in t.__parameters__ if t not in tvars)
```

Comment:
It's actually usually faster to extend from a list comprehension than a generator comprehension. So I'd undo this one.